tests/src/test/scala/org/http4s/UriSpec.scala

```diff
@@ -84,6 +86,44 @@ http://example.org/a file
       }
     }
 
+    "fail to parse port" >> {
+      "if it's negative" >> prop { negative: Int =>
+        {
+          val uri: ParseResult[Uri] = Uri.fromString(s"http://localhost:$negative/")
+          uri match {
+            case Left(ParseFailure("Invalid URI", _)) => ok
+            case unexpected => ko(unexpected.toString)
+          }
+        }
+      }.setGen(Gen.choose[Int](Int.MinValue, -1))
+      "if it's larger than Int.MaxValue" >> prop { tooBig: Long =>
+        {
+          val uri: ParseResult[Uri] = Uri.fromString(s"http://localhost:$tooBig/")
+          uri match {
+            case Left(ParseFailure("Invalid URI", _)) => ok
+            case unexpected => ko(unexpected.toString)
+          }
+        }
+      }.setGen(Gen.choose[Long]((Int.MaxValue: Long) + 1, Long.MaxValue))
+      "if it's not a number or an empty String" >> prop { notNumber: String =>
+        {
+          val uri: ParseResult[Uri] = Uri.fromString(s"http://localhost:$notNumber/")
+          uri match {
+            case Left(ParseFailure("Invalid URI", _)) => ok
+            case unexpected => ko(unexpected.toString)
+          }
+        }
+      }.setGen(Gen.alphaNumStr.suchThat(str =>
+        str.nonEmpty && Either.catchOnly[NumberFormatException](str.toInt).isLeft))
+      "for an empty String" in {
```

Comment:
Locally, this test failed:

```
[info]   fail to parse port
[info]     + if it's negative (214 ms)
[info]     + if it's larger than Int.MaxValue (230 ms)
[info]     + if it's not a number or an empty String (256 ms)
[error]     x for an empty String (271 ms)
[error]      Right(http://localhost/) (UriSpec.scala:121)
[error] org.http4s.UriSpec.$anonfun$new$51(UriSpec.scala:121)
```

@rossabaker. 

Per https://tools.ietf.org/html/rfc3986#page-22, 

```
3.2.3.  Port

   The port subcomponent of authority is designated by an optional port
   number in decimal following the host and delimited from it by a
   single colon (":") character.

      port        = *DIGIT

   A scheme may define a default port.  For example, the "http" scheme
   defines a default port of "80", corresponding to its reserved TCP
   port number.  The type of port designated by the port number (e.g.,
   TCP, UDP, SCTP) is defined by the URI scheme.  URI producers and
   normalizers should omit the port component and its ":" delimiter if
   port is empty or if its value would be the same as that of the
   scheme's default.
```

Per https://tools.ietf.org/html/rfc3986#page-22, 

```
6.2.3.  Scheme-Based Normalization

   The syntax and semantics of URIs vary from scheme to scheme, as
   described by the defining specification for each scheme.
   Implementations may use scheme-specific rules, at further processing
   cost, to reduce the probability of false negatives.  For example,
   because the "http" scheme makes use of an authority component, has a
   default port of "80", and defines an empty path to be equivalent to
   "/", the following four URIs are equivalent:

      http://example.com
      http://example.com/
      http://example.com:/
      http://example.com:80/
```

this test, it seems clear to me, should fail, not pass. In other words, I should replace this "should fail" test with a "should parse to port 80" one.

Please let me know if you disagree.

Reply:
Yup, That seems correct via the spec.